lua/new_side.lua

```diff
@@ -0,0 +1,26 @@
+--[=[
+[new_side]
+Author: MadMax (username on the Battle for Wesnoth forum)
+
+Creates a new side (next available side number) with stated parameters. Accepts side parameters available to [modify_side]. Will not create a leader unit; need to do this yourself with [unit] and canrecruit=yes.
+
+NOTE: manipulating the side's flag crashes Wesnoth as of 1.17.24 (vanilla bug also part of [modify_side]).
```

Comment:
Is there a bug opened about this against upstream Wesnoth that we could cross-reference?

Reply:
@MadMax3947 any further info to share on this one?

Reply:
Just opened one, https://github.com/wesnoth/wesnoth/issues/8193

Reply:
ok, thanks

Reply:
so looking at the issues with `wesnoth.sides.create()` and given there's been talk about removing it, I implemented the revealed dunefolk side differently in https://github.com/cooljeanius/Flight_Freedom/pull/18/commits/eacd3d54026243361e89deb6f3665b4527a93d54